geomstats/information_geometry/fisher_rao_metric.py

```diff
@@ -158,35 +158,34 @@ def pdf(x):
         def _function_to_integrate(x):
             pdf_x = pdf(x)
             pdf_x_at_base_point = pdf_x(base_point)
-            pdf_x_derivative = gs.autodiff.jacobian(pdf_x)
+            pdf_x_derivative = gs.autodiff.jacobian_vec(pdf_x)
             pdf_x_derivative_at_base_point = pdf_x_derivative(base_point)
-            pdf_x_hessian_at_base_point = gs.autodiff.jacobian(pdf_x_derivative)(
-                base_point
-            )
-            return (
-                1
-                / (pdf_x_at_base_point**2)
-                * (
-                    pdf_x_at_base_point
-                    * (
-                        gs.einsum(
-                            "...ki,...j->...ijk",
-                            pdf_x_hessian_at_base_point,
-                            pdf_x_derivative_at_base_point,
-                        )
-                        + gs.einsum(
-                            "...kj,...i->...ijk",
-                            pdf_x_hessian_at_base_point,
-                            pdf_x_derivative_at_base_point,
-                        )
-                    )
-                    - gs.einsum(
-                        "...i, ...j, ...k -> ...ijk",
-                        pdf_x_derivative_at_base_point,
-                        pdf_x_derivative_at_base_point,
+            pdf_x_hessian = gs.autodiff.hessian_vec(pdf_x)
+            pdf_x_hessian_at_base_point = pdf_x_hessian(base_point)
+
+            return gs.einsum(
+                "...n,...ijk->...ijk",
+                1 / (pdf_x_at_base_point**2),
+                gs.einsum(
+                    "...n,...ijk->...ijk",
+                    pdf_x_at_base_point,
+                    gs.einsum(
+                        "...ki,...nj->...ijk",
+                        pdf_x_hessian_at_base_point,
                         pdf_x_derivative_at_base_point,
                     )
+                    + gs.einsum(
+                        "...kj,...ni->...ijk",
+                        pdf_x_hessian_at_base_point,
+                        pdf_x_derivative_at_base_point,
+                    ),
                 )
+                - gs.einsum(
+                    "...ni, ...nj, ...nk -> ...ijk",
+                    pdf_x_derivative_at_base_point,
+                    pdf_x_derivative_at_base_point,
+                    pdf_x_derivative_at_base_point,
+                ),
```

Comment:
looks pretty good. in fact it help to have defined well the output shapes of `pdf`.

later we can improve  this part of the code in terms of performance by computing the jacobian and the hessian simultaneously (but this is something we do not yet available in the backend).

looking forward to see this applied to higher dimensional sample spaces (I guess this part is mostly ready).